pkg/cloudprovider/fake/cloudprovider.go

```diff
@@ -37,8 +36,15 @@ func (c *CloudProvider) Create(_ context.Context, constraints *v1alpha5.Constrai
 	for i := 0; i < quantity; i++ {
 		name := strings.ToLower(randomdata.SillyName())
 		instance := instanceTypes[0]
-		zone := instance.Zones().Intersection(constraints.Requirements.Zones()).UnsortedList()[0]
-		operatingSystem := instance.OperatingSystems().UnsortedList()[0]
+		var zone, capacityType string
+		for _, o := range instance.Offerings() {
+			if constraints.Requirements.CapacityTypes().Has(o.CapacityType) {
+				if constraints.Requirements.Zones().Has(o.Zone) {
+					zone = o.Zone
+					capacityType = o.CapacityType
```

Comment:
Do you want to break here? Does it matter?

Reply:
Oh shoot yeah, good catch. I mean it ultimately doesn't matter, but yeah I was supposed to. I don't know how I missed that =P